internal/trafficcontrol/tc.go

```diff
@@ -189,9 +199,14 @@ func GetSriovVFs(ifName string) ([]string, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to get netlink %s: %q", vfIfName, err)
 		}
-		vfNames = append(vfNames, vfLink.Attrs().Name)
+		vfInfo := &VFInfo{Name: vfLink.Attrs().Name, Index: vfLink.Attrs().Index,
```

Comment:
I think I see something in the code above that's concerning. Basically looking at the directory structure beneath `/sys/class/net/ifname/device/virtfn*`, you *are* guaranteed to have one `virtfn$i` for each 0 <= i <= sriov_numvfs, so just iterating over that filepath.Glob result is good. However:
- If the VF is configure as a vfio_pci device, there will be no `net` directory at all, and no VF interface name, and that's okay. We can still run the traffic control like setting the vlan and tx queue state via the `ip` command with only the PF iface name and VF index number.
- If the VF is configured as a netdev device, there is a `net` directory, but if that netdev has been moved into another network namespace (for example, if it's been assigned to another pod already), that net directory will be empty and we won't know the name of the device unless we do some clever searching.  Again, thankfully the new VLAN-based way of controlling traffic doesn't care if we have the VF name, just the base PF interface name and the VF index.

Reply:
I've done testing on a different SNO (sno-4 in this lab) that checks exactly that - For VFs in other namespaces, and DPDK VFs, does the "set vlan and state" mechanism work, but it would be great to see this in an end-to-end test soon.  I can help set up the containers and SRIOV objects to make it happen if you like.

Reply:
sure, I've tested latest code with VFs (netdevs) and exists on host net namespace, all the basic unauthenticated, authenticated states and vlan change events and stop monitor cases worked fine. When you have few of `ens3f0` VFs into other net namespaces and DPDK mode, then i can verify this one too.

Reply:
@lack I was able to test VFs on other net namespaces and dpdk mode on sno-5 node itself, here is a commit https://github.com/openshift-kni/eapol-operator/pull/12/commits/3a408484d7a6c0181e827128177b40b214a7ef11 needed to make it work.